CMSgov/design-system
# [WNMGDS-3057] Introduce the Shadow DOM to components that need it

## Summary

The whole reason we were selectively introducing the Shadow DOM in our web components was to fix issues with Angular. This pull request fixes Angular event bindings on nested web components for most of the container-like web components by adding the Shadow DOM to the following web components:

- `<ds-accordion>` and `<ds-accordion-item>`
- `<ds-alert>`
- `<ds-drawer>`
- `<ds-modal-dialog>`
- `<ds-review>`

## How to test

1. You can run the web component unit tests.
2. You can test these in Storybook to make sure they still work as expected.

Note that the updates to the example projects will come in another PR.
